recipes/master.rb

```diff
@@ -20,6 +20,15 @@
   action :enable
 end
 
+user node['salt']['master']['api']['user']['name'] do
```

Comment:
Putting the `user` resource here assumes that someone doesn't already have the needed user. A lot of people, including me, create service users separate with quite different properties listed.  Also, if a user named `saltapi` is already created somewhere else, this will break in Chef13 unless someone explicitly sets the name even if they don't use the resource.

I could see this staying if it was actually put into an if block instead of using an `only_if`.  Then doing the following
```
  node['salt']['master']['api']['user'].tap{ |x| x.delete('enable') }.each do |config, value|
    send(config.to_sym, value) unless value.nil?
  end
```